runfabtests.sh: Fix is_excluded function to only match full test names.

bturrubiates opened the pull request:
Fixes issue where test names that are a substring of an excluded test name get excluded.
Changed to `test_exe` since it includes the `fi` prefix. In the function `$1` will only expand to the first part of `test_exe` which is `fi_test`. 

Example:
```bash
$ EXCLUDE="fi_cq_data,fi_dgram_waitset,fi_msg_poll,fi_poll,fi_cmatose,fi_rdm_cntr_pingpong"
$ function old_is_excluded {
>   local e=$(echo $EXCLUDE | sed -e s/$1//)
>   if [[ "$EXCLUDE" != "$e" ]]; then
>     echo 1
>   else
>     echo 0
>   fi
>   return
> }
$ function new_is_excluded {
>   local i
>   for i in $(echo "$EXCLUDE" | tr -s "," " "); do
>     if [[ "$i" = "$1" ]]; then
>       echo 1
>       return
>     fi
>   done
>   echo 0
> }
$ old_is_excluded fi_dgram
1
$ new_is_excluded fi_dgram
0
$ old_is_excluded fi_dgram_waitset
1
$ new_is_excluded fi_dgram_waitset
1
```

@pmmccorm @goodell @shefty @jsquyres 
Please review.

Signed-off-by: Ben Turrubiates <bturrubi@cisco.com>

pmmccorm commented on scripts/runfabtests.sh:
```diff
@@ -155,13 +155,15 @@ function compute_duration {
-        echo 0
-    fi
-    return
+	local i
```
Nit: we can just omit this "local i" right?

bturrubiates replied:
I'd be fine with it, but everywhere else seemed to use function locals.

pmmccorm commented on scripts/runfabtests.sh:
```diff
@@ -155,13 +155,15 @@ function compute_duration {
+		fi
+	done
+
+	echo 0
```
Nit: why not just return 1 or 0, and then below:

```
if [ is_excluded $test_exe ]; then
...
```

goodell replied:
@pmmccorm I think you would want to omit the `[` `]`, since you don't want to invoke `test` here.  I.e.,
```sh
if is_excluded "$test_exe" ; then
...
```

However, I think we welcome your patch for such a modification after Ben's PR lands, since Ben is just making a straightforward correction of the logic in this function, not changing the way the function is called.

pmmccorm replied:
Sounds good.

pmmccorm commented:
A couple nits but otherwise looks better than the old way.

goodell commented on scripts/runfabtests.sh:
```diff
@@ -173,7 +175,7 @@ function unit_test {
 	local test_time
 
-	local e=$(is_excluded $test)
+	local e=$(is_excluded $test_exe)
```
Doesn't this mean that the "exclude" format will now change to look like this?
```sh
runfabtests.sh [...] -e "fi_ud_pingpong -f usnic,fi_ud_pingpong -P -f usnic"
```

bturrubiates replied:
No, the exclude format shouldn't change. The `$test` string doesn't have the `fi` prefix prepended. The `$test_exe` does have a bunch of stuff after it, but that's not really different since some of the tests have options appended anyway. By passing `$test_exe` and not `"$test_exe"` I'm only getting the first word in the string (test name).

goodell replied:
Ick.  That level of subtlety (deliberately allowing `$test_exe` to expand and be word-split at invocation time so that only `fi_foo` ends up in `$1`) warrants a comment or a change in the logic to make it clear.  More obvious would be to pass `"fi_${test}"` instead of `$test_exe`.

bturrubiates replied:
`"fi_${test}"` isn't right either though. For example, the first test is `fi_av_test` but with the way the test is written if I did `"fi_${test}"` it would expand to `"fi_av_test -d GOOD_ADDR -n 1 -s SERVER_ADDR"` in which case I'd also be pulling the unquoted `$1` trick.

bturrubiates commented:
`"fi_${test}"` isn't right either though. For example, the first test is `fi_av_test` but with the way the test is written if I did `"fi_${test}"` it would expand to `"fi_av_test -d GOOD_ADDR -n 1 -s SERVER_ADDR"` in which case I'd also be pulling the unquoted `$1` trick.

bturrubiates commented:
@pmmccorm 
Removed the `local i`

@goodell 
Tried to make it more clear. Changed using `$test_exe` to using `fi_${test}`. `fi_${test}` still sometimes has extra parameters (as in the case of `av_test`). Now instead we're explicitly passing only the first space delimited field.

goodell commented:
New version is fine enough for me.  The script is starting to show the limits of shell scripting...

:+1:

pmmccorm commented:
Looks good to me.
